harness/tests/integration_cases/test_raft.rs

```diff
@@ -5567,3 +5567,280 @@ fn test_uncommitted_state_advance_ready_from_last_term() {
+            vec![
+                empty_entry(1, 1),
+                empty_entry(1, 2),
+                empty_entry(1, 3),
+                empty_entry(4, 4),
+            ],
+            4,
+            4,
+            4,
+            4,
+        ),
+        // Test case from example comment in stepLeader (on leader).
+        (
+            vec![
+                empty_entry(2, 1),
+                empty_entry(5, 2),
+                empty_entry(5, 3),
+                empty_entry(5, 4),
+                empty_entry(5, 5),
+                empty_entry(5, 6),
+                empty_entry(5, 7),
+                empty_entry(5, 8),
+                empty_entry(5, 9),
+            ],
+            vec![
+                empty_entry(2, 1),
+                empty_entry(4, 2),
+                empty_entry(4, 3),
+                empty_entry(4, 4),
+                empty_entry(4, 5),
+                empty_entry(4, 6),
+            ],
+            4,
+            6,
+            2,
+            1,
+        ),
+        // Test case from example comment in handleAppendEntries (on follower).
+        (
+            vec![
+                empty_entry(2, 1),
+                empty_entry(2, 2),
+                empty_entry(2, 3),
+                empty_entry(2, 4),
+                empty_entry(2, 5),
+            ],
+            vec![
+                empty_entry(2, 1),
+                empty_entry(4, 2),
+                empty_entry(4, 3),
+                empty_entry(4, 4),
+                empty_entry(4, 5),
+                empty_entry(4, 6),
+                empty_entry(4, 7),
+                empty_entry(4, 8),
+            ],
+            2,
+            1,
+            2,
+            1,
+        ),
+    ];
+    for (
+        i,
+        (
+            leader_log,
+            follower_log,
+            reject_hint_term,
+            reject_hint_index,
+            next_append_term,
+            next_append_index,
+        ),
+    ) in tests.drain(..).enumerate()
+    {
+        // TODO(accelsao): maybe we need `Copy` of Message instead of recreate by new_message
+        let l = default_logger();
+        let s1 = MemStorage::new_with_conf_state((vec![1, 2, 3], vec![]));
+        s1.wl().append(&leader_log).unwrap();
+        let s2 = MemStorage::new_with_conf_state((vec![1, 2, 3], vec![]));
+        s2.wl().append(&follower_log).unwrap();
+        let mut n1 = new_test_raft(1, vec![1, 2, 3], 10, 1, s1, &l);
+        let mut n2 = new_test_raft(2, vec![1, 2, 3], 10, 1, s2, &l);
+        n1.become_candidate();
+        n1.become_leader();
+        n2.step(new_message(2, 2, MessageType::MsgHeartbeat, 0))
+            .unwrap();
+        let msgs = n2.read_messages();
+        assert_eq!(msgs.len(), 1, "#{}", i);
+        assert_eq!(
+            msgs[0].get_msg_type(),
+            MessageType::MsgHeartbeatResponse,
+            "#{}",
+            i
+        );
+        n1.step(new_message(2, 2, MessageType::MsgHeartbeatResponse, 0))
```

Comment:
Why not just `n1.step(msgs[0])`?

Reply:
No Copy for Message, I am not familiar with proto custom implement yet

Reply:
If `msgs` is a vec, then it's move instead of copy. Besides, copy works the same as recreate the message and more readable.

Reply:
But the error says it is because of `Copy`
```
error[E0507]: cannot move out of index of `Vec<raft::prelude::Message>`
    --> harness/tests/integration_cases/test_raft.rs:5811:17
     |
5811 |         n1.step(msgs[0]).unwrap();
     |                 ^^^^^^^ move occurs because value has type `raft::prelude::Message`, which does not implement the `Copy` trait

error: aborting due to previous error
```

Reply:
I found that I can use `pop()` for a workaround